main.cpp

```diff
@@ -112,20 +177,27 @@ struct SyncShift
     //
     ///////////////////////////////////////////////////////////
 
-    int setKnob1(int k1) { knob1 = k1; return knob1; } 
-    int setKnob2(int k2) { knob2 = k2; return knob2; } 
-    int setKnob3(int k3) { knob1 = k3; return knob3; } 
+    int setKnob1(int k1);  
+    int setKnob2(int k2);  
+    int setKnob3(int k3);  
 
     ///////////////////////////////////////////////////////////
     //
     // setknob 4 function call return new knob value   
     //
     ///////////////////////////////////////////////////////////   
 
-    double setKnob4(double k4) { knob4 = k4; return knob4; }
+    double setKnob4(double k4);
 
 };  // end Struct SyncShift
-
+// implement SyncShift::setKnob1
+int SyncShift::setKnob1(int k1) { knob1 = k1; return knob1; } 
+// implement SyncShift::setKnob2
+int SyncShift::setKnob2(int k2) { knob2 = k2; return knob2; }
+// implement SyncShift::setKnob3
+int SyncShift::setKnob3(int k3) { knob1 = k3; return knob3; }
+// implement SyncShift::setKnob4
+double SyncShift::setKnob4(double k4) { knob4 = k4; return knob4; }
```

Comment:
Do you need all of these comments?   They just clutter up the code.   the function names are self-explanatory.